src/sentry/tasks/summaries/weekly_reports.py

```diff
@@ -129,9 +129,9 @@ def prepare_organization_report(
     timestamp: float,
     duration: int,
     organization_id: int,
-    batch_id: uuid.UUID,
+    batch_id: uuid.UUID | str,
     dry_run: bool = False,
-    target_user: User | None = None,
+    target_user: User | int | None = None,
     email_override: str | None = None,
 ):
     if target_user and not hasattr(target_user, "id"):
```

Comment:
Wouldn't this condition pass and start dropping activations once we start sending `target_user` as an int?

The callsites that send this parameter are all in `getsentry`. We'll need to not change the callers until this change has deployed to avoid new activations being processed by old code.

Reply:
Good catch, that's true. Updated the condition to also check the param type.